readme.md

```diff
@@ -540,24 +540,23 @@ let map = (f) => (list) => list.map(f)
 ```
 
 ## Union type
-A union type is the combination of two types together into another one.
 
-JS doesn't have static types but let's say we invent a type `NumOrString` which is a sum of `String` and `Number`.
+A **union** type is a union across a set of other types.
 
-The `+` operator in JS works on strings and numbers so we can use this new type to describe its inputs and outputs:
+We can think of each type as a set of possible values. (This is the set-theoretic model of types.) A union across two types is another set of values, where each element in the union is an element of one of the component sets.
 
-```js
-// add :: (NumOrString, NumOrString) -> NumOrString
-const add = (a, b) => a + b;
+JavaScript doesn't have static types but let's say we invent a type `NumberOrString` which is a union of `Number` and `String`.
 
-add(1, 2); // Returns number 3
-add('Foo', 2); // Returns string "Foo2"
-add('Foo', 'Bar'); // Returns string "FooBar"
+```js
+var s = new String('abc')
+var n = new Number(123)
+var nos = new NumberOrString(123)
```

Comment:
I'm not sure you need a specialized constructor to say that a value is of a type. I'd also like us to continue to have an example that shows how union types work with function definitions.